lib/src/async_io.rs

```diff
@@ -7,28 +7,28 @@ use {
     futures::{FutureExt, TryFutureExt},
     std::time::Duration,
     tokio::time::timeout,
-    wiggle::GuestPtr,
+    wiggle::{GuestMemory, GuestPtr},
 };
 
 #[wiggle::async_trait]
 impl FastlyAsyncIo for Session {
-    async fn select<'a>(
+    async fn select(
         &mut self,
-        handles: &GuestPtr<'a, [AsyncItemHandle]>,
+        memory: &mut GuestMemory<'_>,
+        handles: GuestPtr<[AsyncItemHandle]>,
         timeout_ms: u32,
     ) -> Result<u32, Error> {
-        let handles = GuestPtr::<'a, [u32]>::new(handles.mem(), handles.offset())
-            .as_slice()?
-            .ok_or(Error::SharedMemory)?;
+        let handles = handles.cast::<[u32]>();
         if handles.len() == 0 && timeout_ms == 0 {
             return Err(Error::InvalidArgument);
         }
 
         let select_fut = self
             .select_impl(
-                handles
-                    .iter()
-                    .copied()
+                memory
+                    // TODO: wiggle only supports slices of u8 in 22.0.0
+                    .to_vec(handles)?
+                    .into_iter()
                     .map(|i| AsyncItemHandle::from(i).into()),
```

Comment:
Is there a limitation implied here, like we can only support 256 async item handles rather than the full u32 range of handles previously supported?

Reply:
OH, is the limitation that we can only have _slices_ of u8s, but we can have full `Vec`s of other scalar types?

Reply:
Yep, that's exactly it. I'm going to fix wiggle so that we can have slices of other things, and this comment was mostly a reminder to me that this should be updated to not copy in the future.